### PR TITLE
Allow protocol in --ci arg, optionally disable SSL verification

### PR DESCRIPTION
**Pull request by manics**
`--ci {,http://,https://}ci.openmicroscopy.org --branch XXXX` should all work

If you are downloading from the latest devspace with a self-signed SSL certificate you can disable verification: `OMEGO_SSL_NO_VERIFY=1 omego download --ci https://1.2.3.4 --branch OMERO-build`

- Fixes https://github.com/ome/omego/issues/79
- Partially fixes https://github.com/ome/omego/issues/73
- The name `OMEGO_SSL_NO_VERIFY` is based on `GIT_SSL_NO_VERIFY` https://blog.breadncup.com/2011/06/09/skip-git-ssl-verification/

Tests are still broken due to https://github.com/ome/omego/issues/89

### CODE REVIEW

**Comment by manics** [comment]
I think this also fixes https://github.com/ome/omego/issues/82

**Comment by manics** [comment]
@pwalczysko This isn't ready for a proper review because the tests are still broken, but you're welcome to test it- it fixes #79.

**Comment by manics** [comment]
Travis fixed thanks to #91

File: omego/fileutils.py [hunk 1]
```diff
@@ -42,14 +43,26 @@ def update(self, current):
...
       neither must be provided)
+    method: The HTTP method
+
+    Caller is reponsible for calling close() on the returned object
     """
-    opener = urllib2.build_opener()
+    if os.getenv('OMEGO_SSL_NO_VERIFY') == '1':
+        # This needs to come first to override the default HTTPS handler
+        log.debug('OMEGO_SSL_NO_VERIFY=1')
+        sslctx = ssl.create_default_context()
```
Comment (sbesson) [inline omego/fileutils.py]: For me this failed with 
```
Traceback (most recent call last):
  File "omego/main.py", line 62, in <module>
    entry_point()
  File "omego/main.py", line 52, in entry_point
    (Version.NAME, Version)])
  File "/fc/homes/sbesson/.local/lib/python2.6/site-packages/yaclifw/framework.py", line 197, in main
    ns.func(ns)
  File "/fc/homes/sbesson/omego2/omego/artifacts.py", line 443, in __call__
    artifacts = Artifacts(args)
  File "/fc/homes/sbesson/omego2/omego/artifacts.py", line 39, in __init__
    self.artifacts = JenkinsArtifacts(args)
  File "/fc/homes/sbesson/omego2/omego/artifacts.py", line 210, in __init__
    root = self.read_xml(buildurl)
  File "/fc/homes/sbesson/omego2/omego/artifacts.py", line 229, in read_xml
    url = fileutils.open_url(buildurl + 'api/xml')
  File "/fc/homes/sbesson/omego2/omego/fileutils.py", line 59, in open_url
    sslctx = ssl.create_default_context()
```

**Comment by sbesson** [comment]
Apart from the `OMEGO_SSL_NO_VERIFY`  issue reported above, edownloading artifacts from devspaces with HTTPS, custom port using the `--ci` argument now works as expected.

File: omego/fileutils.py (hunk 1 above)
Comment (manics) [inline omego/fileutils.py]: Is that the full error message? Did `create_default_context` fail, or does it not exist?

Comment (sbesson) [inline omego/fileutils.py]: It seems to not exist

```
>>> import ssl
>>> ssl.create_default_context()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'create_default_context'
```

**Comment by manics** [comment]
@sbesson I've added a `try`-`except`-`Stop` wrapper

**Comment by sbesson** [comment]
Retested with the latest commit

```
$ OMEGO_SSL_NO_VERIFY=1 python omego/main.py download --ci https://1.2.3.4:4443 --branch OMERO-build matlab -v
2016-06-29 19:38:22,367 [omego.artifa] DEBUG             artifact => matlab
2016-06-29 19:38:22,367 [omego.artifa] DEBUG               branch => OMERO-build
2016-06-29 19:38:22,367 [omego.artifa] DEBUG                   ci => https://1.2.3.4:4443
2016-06-29 19:38:22,368 [omego.artifa] DEBUG          downloadurl => http://downloads.openmicroscopy.org
2016-06-29 19:38:22,368 [omego.artifa] DEBUG            overwrite => keep
2016-06-29 19:38:22,368 [omego.artifa] DEBUG buildurl: https://1.2.3.4:4443/job/OMERO-build/lastSuccessfulBuild/
2016-06-29 19:38:22,368 [omego.fileut] DEBUG OMEGO_SSL_NO_VERIFY=1
2016-06-29 19:38:22,368 [omego.fileut] ERROR Failed to create Default SSL context: 'module' object has no attribute 'create_default_context'
ERROR: 
Failed to create Default SSL context, OMEGO_SSL_NO_VERIFY is not supported on older versions of Python
```

**Pull request merged**